.github/actions/ghasum/action.yml

```diff
@@ -0,0 +1,185 @@
+      env:
+        GH_TOKEN: ${{ github.token }}
+        VERSION: ${{ inputs.version }}
+      run: |
+        gh release download "$VERSION" --repo chains-project/ghasum --pattern "$ARTIFACT"
+        shasum --check --ignore-missing checksums-sha512.txt
+        tar -xf "$ARTIFACT"
+    - name: Verify the action checksums
+      if: runner.os == 'Linux'
+      shell: bash
+      env:
+        JOB: ${{ github.job }}
+        WORKFLOW: ${{ github.workflow_ref }}
+      run: |
+        WORKFLOW=$(echo "$WORKFLOW" | cut -d '@' -f 1 | cut -d '/' -f 3-5)
+        /tmp/ghasum/ghasum verify -cache /home/runner/work/_actions -no-evict -offline "$WORKFLOW:$JOB"
+
+    # Windows
+    - name: Pick the ghasum CLI (amd64)
+      if: runner.os == 'Windows' && runner.arch == 'X64'
+      shell: pwsh
+      run: |
+        'ARTIFACT=ghasum_windows_amd64.zip' >>"$env:GITHUB_ENV"
+    - name: Pick the ghasum CLI (arm64)
+      if: runner.os == 'Windows' && runner.arch == 'ARM64'
+      shell: pwsh
+      run: |
+        'ARTIFACT=ghasum_windows_arm64.zip' >>"$env:GITHUB_ENV"
+    - name: Download the ghasum CLI
+      if: runner.os == 'Windows'
+      shell: pwsh
+      working-directory: C:\ghasum
+      env:
+        GH_TOKEN: ${{ github.token }}
+        VERSION: ${{ inputs.version }}
+      run: |
+        gh release download "$env:VERSION" --repo chains-project/ghasum --pattern "$env:ARTIFACT"
+        $line = Get-Content checksums-sha512.txt | Where-Object { $_ -match "\b$env:ARTIFACT$" }
+        if (-not $line) {
+          Write-Error 'Checksum missing'
+          exit 2
+        } else {
+          if ($line -match "^([a-fA-F0-9]+)  $env:ARTIFACT$") {
+            $want = $matches[1]
+            $got = (Get-FileHash -Path $env:ARTIFACT -Algorithm SHA512).Hash
+            if ($got.ToLower() -ne $want.ToLower()) {
+              Write-Error 'Checksum mismatch'
+              exit 1
+            } else {
+              Write-Host 'Checksum match'
+              Expand-Archive -Path "$env:ARTIFACT" -DestinationPath .
+            }
+          } else {
+            Write-Error 'Checksums malformed'
+            exit 2
+          }
+        }
+    - name: Verify the action checksums
+      if: runner.os == 'Windows'
+      shell: pwsh
+      env:
+        JOB: ${{ github.job }}
+        WORKFLOW: ${{ github.workflow_ref }}
+      run: |
+        $WorkflowParts = $env:WORKFLOW -split '@'
+        $WorkflowPath = ($WorkflowParts[0] -split '/')[2..4] -join '/'
+        if (Test-Path -Path 'C:\a\_actions') {
+          C:\ghasum\ghasum.exe verify -cache C:\a\_actions -no-evict -offline "${WorkflowPath}:$env:JOB"
+        } else {
+          C:\ghasum\ghasum.exe verify -cache D:\a\_actions -no-evict -offline "${WorkflowPath}:$env:JOB"
+        }
+
+    # Cleanup
+    - name: Cleanup (Unix)
+      if: runner.os == 'macOS' || runner.os == 'Linux'
+      shell: bash
+      run: |
+        rm -rf /tmp/ghasum
+        echo 'ARTIFACT=' >>"$GITHUB_ENV"
```

Comment:
Unfortunately this does not remove the environment variable (ref. ac9b9dce64b1bcdbb8dbc2a16562e93b104437bc). If possible I would really like to avoid polluting the environment variables for subsequent steps :thinking:

Reply:
Resolved with a337d8bff379c30046d145ce348795fcda612d63+4f341211731734b1d18fb81c6ac6e118a92d88c6